pkg/statsd/handler_cloud.go

```diff
@@ -41,6 +50,92 @@ type CacheOptions struct {
 	CacheNegativeTTL          time.Duration
 }
 
+// CloudHandlerFactory creates a CloudHandler based on some predefined options.
+type CloudHandlerFactory struct {
+	cloudProviderName string
+	version           string
+	CloudProvider     gostatsd.CloudProvider
+	logger            logrus.FieldLogger
+	Limiter           *rate.Limiter
+	CacheOptions      *CacheOptions
```

Comment:
This doesn't need to be a pointer, better as just a value.
Also all fields should be private OR make them all public and remove the constructor, I guess.